File: HashMaps_problems/find-players-with-zero-or-one-losses.cpp
class Solution {
public:
    vector<vector<int>> findWinners(vector<vector<int>>& matches) {
        
       unordered_map<int,int>up;
       unordered_set<int>seen;

       for(auto & match:matches)
       {
           int w = match[0],los = match[1];
           seen.insert(w);
           seen.insert(los);
           up[los]++;

       }

       vector<vector<int>>res(2,vector<int>());

       for(auto p:seen)
       {
           if(up.find(p) == up.end())
           {
               res[0].push_back(p);
           }
           else if(up[p] == 1)
           {
               res[1].push_back(p);
           }
       }

       sort(res[0].begin(),res[0].end());
       sort(res[1].begin(),res[1].end());

       return res;
       
    }
};

File: HashMaps_problems/find-players-with-zero-or-one-losses.py
class Solution:
    def findWinners(self, matches: List[List[int]]) -> List[List[int]]:
        
        match = {}
        
        for i in range(len(matches)):
            winner = matches[i][0]
            loser = matches[i][1]
            
            if winner not in match:
                match[winner] = [0,0]
                match[winner][0] += 1
            elif winner in match:
                match[winner][0] += 1
            if loser not in match:
                match[loser] = [0,0]
                match[loser][1] += 1
            elif loser in match:
                match[loser][1] += 1
        win = []
        loss = []
        #print(match)
        for player,data in match.items():
            if data[1] == 0:
                win.append(player)
            elif data[1] == 1:
                loss.append(player)
        win.sort()
        loss.sort()
        return [win,loss]
        
